test/test.auth.js

```
@@ -1,59 +1,96 @@
+         github = new Github({
+            username: testUser.USERNAME,
+            password: testUser.PASSWORD,
+            auth: 'basic'
+         });
+
+         user = github.getUser();
       });
 
-      user = github.getUser();
-   });
+      // 200ms between tests so that Github has a chance to settle
+      beforeEach(function(done) {
+         setTimeout(done, 200);
+      });
 
-   it('should authenticate and return no errors', function(done) {
-      user.notifications(function(err) {
-         should.not.exist(err);
-         done();
+      it('should authenticate and return no errors', function(done) {
+         user.notifications(assertSuccessful(done));
       });
    });
-});
 
-describe('Github constructor without authentication data', function() {
-   it('should read public information', function(done) {
-      var github = new Github();
-      var gist = github.getGist('f1c0f84e53aa6b98ec03');
+   describe('without authentication', function() {
+      before(function() {
+         github = new Github();
+      });
+
+      // 200ms between tests so that Github has a chance to settle
+      beforeEach(function(done) {
+         setTimeout(done, 200);
+      });
+
+      it('should read public information', function(done) {
+         var gist = github.getGist('f1c0f84e53aa6b98ec03');
+
+         gist.read(function(err, res, xhr) {
+            try {
+               expect(err).not.to.exist();
+               expect(res).to.exist();
+               expect(xhr).to.be.an.object();
+
+               done();
+            } catch(e) {
+               try {
+                  if (err && err.request.headers['x-ratelimit-remaining'] === '0') {
+                     done();
 
-      gist.read(function(err, res, xhr) {
-         should.not.exist(err);
-         xhr.should.be.instanceof(XMLHttpRequest);
-         res.should.be.an('object');
+                     return;
+                  }
+               } catch(e2) {
+                  done(e);
```

I think this line can be avoided an have an empty catch.